examples/keras_io/vision/metric_learning.py

```diff
@@ -0,0 +1,321 @@
+"""
+Title: Metric learning for image similarity search
+Author: [Mat Kelcey](https://twitter.com/mat_kelcey)
+Date created: 2020/06/05
+Last modified: 2020/06/09
+Description: Example of using similarity metric learning on CIFAR-10 images.
+Accelerator: GPU
+"""
+"""
+## Overview
+
+Metric learning aims to train models that can embed inputs into a high-dimensional space
+such that "similar" inputs, as defined by the training scheme, are located close to each
+other. These models once trained can produce embeddings for downstream systems where such
+similarity is useful; examples include as a ranking signal for search or as a form of
+pretrained embedding model for another supervised problem.
+
+For a more detailed overview of metric learning see:
+
+* [What is metric learning?](http://contrib.scikit-learn.org/metric-learn/introduction.html)
+* ["Using crossentropy for metric learning" tutorial](https://www.youtube.com/watch?v=Jb4Ewl5RzkI)
+"""
+
+"""
+## Setup
+"""
+
+import random
+import matplotlib.pyplot as plt
+import numpy as np
+import tensorflow as tf
+from collections import defaultdict
+from PIL import Image
+from sklearn.metrics import ConfusionMatrixDisplay
+import os
+import keras
+from keras import layers
+
+"""
+## Dataset
+
+For this example we will be using the
+[CIFAR-10](https://www.cs.toronto.edu/~kriz/cifar.html) dataset.
+"""
+
+from keras.datasets import cifar10
+
+"""
+## Backend for the new keras
+
+Set Keras backend to tensorflow. 
+"""
+os.environ["KERAS_BACKEND"] = "tensorflow"
```

Comment:
the backend should be set before `import keras`, so I suggest adding it under the setup section

Reply:
done.